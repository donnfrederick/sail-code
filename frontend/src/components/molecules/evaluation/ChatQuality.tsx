import * as React from 'react'

import Checkbox from 'components/atoms/evaluation/Checkbox'
import questionnaireHoc from 'components/molecules/evaluation/questionnaireHoc'
import {
  chatQualityChoices,
  ComponentProps,
  QualityUnion,
  QuestionnaireChoice,
  StateHook
} from 'models/evaluations'

interface Props extends ComponentProps {
  chatQualityState: StateHook<number[]>
  onSubmit: () => void
}

export default ({
  chatQualityState,
  stepState,
  isTeacher,
  onSubmit
}: Props) => {
  const [chatQuality, setChatQuality] = chatQualityState
  const onSelect = (choice: QualityUnion) =>
    setChatQuality([...chatQuality, choice])
  const onUnselect = (choice: QualityUnion) =>
    setChatQuality(chatQuality.filter(i => i !== choice))
  const chatQualityChoice = isTeacher
    ? chatQualityChoices.teacher
    : chatQualityChoices.student
  const ChatQuality = () => (
    <>
      {chatQualityChoice.map((choice: QuestionnaireChoice<QualityUnion>) => (
        <Checkbox
          key={choice.value}
          text={choice.text}
          isActive={chatQuality.includes(choice.value)}
          onSelect={() => onSelect(choice.value)}
          onUnselect={() => onUnselect(choice.value)}
        />
      ))}
    </>
  )
  const QuestionnaireEl = questionnaireHoc(ChatQuality)
  return (
    <QuestionnaireEl
      stepState={stepState}
      isActive={true}
      headerWidth={640}
      subHeaderAlign="left"
      isTeacher={isTeacher}
      onSubmit={onSubmit}
    />
  )
}
