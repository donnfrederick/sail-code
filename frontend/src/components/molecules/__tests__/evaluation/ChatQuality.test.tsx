import ChatQuality from 'components/molecules/evaluation/ChatQuality'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

test('render <ChatQuality />', () => {
  const Tree = () => {
    const isTeacher = true
    const stepState = React.useState<number>(4)
    const chatQualityState = React.useState<number[]>([])
    return (
      <ChatQuality
        chatQualityState={chatQualityState}
        isTeacher={isTeacher}
        stepState={stepState}
        onSubmit={() => console.log('submit')}
      />
    )
  }
  expect(renderer.create(<Tree />).toJSON()).toMatchSnapshot()
})
