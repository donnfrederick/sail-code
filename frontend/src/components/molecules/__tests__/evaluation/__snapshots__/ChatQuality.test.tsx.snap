// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <ChatQuality /> 1`] = `
Array [
  <div
    className="sc-gzVnrw kRftEK"
  >
    <div
      className="sc-jTzLTM biVWil"
    >
      <div
        className="sc-jzJRlG kGjMcW"
      />
      <div
        className="sc-fjdhpX ldPbZJ"
        data-completed={true}
        data-current={false}
      >
        1
      </div>
      <div
        className="sc-fjdhpX ldPbZJ"
        data-completed={true}
        data-current={false}
      >
        2
      </div>
      <div
        className="sc-fjdhpX ldPbZJ"
        data-completed={true}
        data-current={false}
      >
        3
      </div>
      <div
        className="sc-fjdhpX ldPbZJ"
        data-completed={true}
        data-current={true}
      >
        4
      </div>
    </div>
    <div
      className="sc-ifAKCX cnjyrT"
      width={640}
    >
      <h2
        className="sc-EHOje hkpgLc"
      >
        映像や音声の品質はどうでしたか？
      </h2>
      <h3
        className="sc-bZQynM jcLHie"
      >
        会話した方の音声や周囲の環境（雑音など）、映像はいかがでしたか？
問題ない場合はそのまま終わりを押してください。
※ 複数選択可
      </h3>
    </div>
  </div>,
  <div
    className="sc-dnqmqq gEEnOm"
  >
    <div
      className="sc-bdVaJa cXZfnx"
      onClick={[Function]}
    >
      <div
        className="sc-bwzfXH kzjiuI"
        data-active={false}
      >
        <img
          className="sc-htpNat ldDEDY"
          data-active={false}
          src="/assets/img/common/evaluation/check@2x.png"
        />
      </div>
      <span
        className="sc-bxivhb jYwlOI"
      >
        映像がたまに切れてた
      </span>
    </div>
    <div
      className="sc-bdVaJa cXZfnx"
      onClick={[Function]}
    >
      <div
        className="sc-bwzfXH kzjiuI"
        data-active={false}
      >
        <img
          className="sc-htpNat ldDEDY"
          data-active={false}
          src="/assets/img/common/evaluation/check@2x.png"
        />
      </div>
      <span
        className="sc-bxivhb jYwlOI"
      >
        まったく映らなかった
      </span>
    </div>
    <div
      className="sc-bdVaJa cXZfnx"
      onClick={[Function]}
    >
      <div
        className="sc-bwzfXH kzjiuI"
        data-active={false}
      >
        <img
          className="sc-htpNat ldDEDY"
          data-active={false}
          src="/assets/img/common/evaluation/check@2x.png"
        />
      </div>
      <span
        className="sc-bxivhb jYwlOI"
      >
        周囲の雑音が多かった
      </span>
    </div>
    <div
      className="sc-bdVaJa cXZfnx"
      onClick={[Function]}
    >
      <div
        className="sc-bwzfXH kzjiuI"
        data-active={false}
      >
        <img
          className="sc-htpNat ldDEDY"
          data-active={false}
          src="/assets/img/common/evaluation/check@2x.png"
        />
      </div>
      <span
        className="sc-bxivhb jYwlOI"
      >
        音声がブツブツ切れてた
      </span>
    </div>
  </div>,
  <div
    className="sc-htoDjs bqFwbt"
  >
    <button
      className="sc-cSHVUG fBbgMe"
      data-active={true}
      fontSize={32}
      height={88}
      onClick={[Function]}
      type="blue"
      width={360}
    >
      終わり
    </button>
  </div>,
]
`;
